File: RailwayReservation_BackEnd/src/main/java/com/edubridge/service/TransactionService.java
package com.edubridge.service;

import java.util.List;

import com.edubridge.entity.Transaction;

public interface TransactionService 
{
	public Transaction addtransaction(Transaction transaction, long accountId, long customerId);
	public Transaction getTransactionById(long transactionId);
	public Transaction getTransactionByAccountId(long accountId);
	public List<Transaction> getTransactionByCustomerId(long customerId);
}

File: RailwayReservation_FrontEnd/src/app/services/transaction.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Transaction } from '../pages/module/transaction';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class TransactionService {
  constructor(private http:HttpClient) { }
  private url="http://localhost:8082/api/transaction"

public addtransaction(transaction:Transaction,customerId:number,accountId:number):Observable<any>
{
return this.http.post<any>(`${this.url}/${customerId}/${accountId}`,transaction);
}
public getTransactionById(transactionId:number)
{
return this.http.get<any>(`${this.url}/${transactionId}`)
}
public getTransactionByAccountId(accountId:number):Observable<any>
{
return this.http.get<any>(`${this.url}/account/${accountId}`)
}
public getTransactionByCustomerId(customerId:number):Observable<any>
{
return this.http.get<any>(`${this.url}/customer/${customerId}`)
}
}
